src/Http/Client/Response.php

```diff
@@ -557,9 +556,14 @@ public function version()
      * @param callable|null $parser The callback to use to decode
      *   the response body.
      * @return mixed The response body.
+     * @deprecated 3.7.0 Use getStringBody()/getJson()/getXml() instead.
      */
     public function body($parser = null)
     {
+        deprecationWarning(
+            'Response::body() is deprecated. Use getStringBody()/getJson()/getXml() instead.'
+        );
```

Comment:
How should people replace parsed bodies? Should we have `getParsedBody($func)`?

Reply:
Do we really need the ability to pass parsers as argument? New methods are already added to getting JSON and XML parses content which are most common types.

People can just pass the return value of `$response->getStringBody()` to their custom parser functions instead.

Reply:
> People can just pass the return value of $response->getStringBody() to their custom parser functions instead.

That's fair.